src/pages/experiments/[experimentId]/plots-and-tables/volcano/index.jsx

```diff
@@ -203,12 +218,11 @@ const VolcanoPlot = ({ experimentId }) => {
       );
     }
 
-    if (plotData.length === 0 || loading || _.isEmpty(spec.spec)) {
+    if (plotData.length === 0 || diffExprLoading) {
       return <Loader experimentId={experimentId} />;
     }
-    return (
-      <Vega data={{ data: plotData }} spec={spec.spec} renderer='canvas' />
-    );
+
+    return <Vega spec={generateSpec(config, plotData)} renderer='canvas' />;
```

Comment:
Is it good to call `generateSpecz on every re-render? Does it slow things down more?

Reply:
Good point. I've wrapped the call to `generateSpec` around a use effect and a check to see if the config is still the same. It initially rendered the plot 3 times. After that, it renders the plot only once.